hypothesis-python/tests/cover/test_baseexception.py

```diff
@@ -0,0 +1,77 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Most of this work is copyright (C) 2013-2019 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import absolute_import, division, print_function
+
+import pytest
+
+from hypothesis import given
+from hypothesis.strategies import composite, integers
+
+
+def test_keyboardinterrupt_no_rerun():
+    runs = [0]
+    interrupt = 3
+
+    @given(integers())
+    def test_raise_keyboardinterrupt(x):
+        runs[0] += 1
+        if runs[0] == interrupt:
+            raise KeyboardInterrupt
+
+    with pytest.raises(KeyboardInterrupt):
+        test_raise_keyboardinterrupt()
+
+    assert runs[0] == interrupt
+
+
+def test_keyboardinterrupt_in_strategy_no_rerun():
+    runs = [0]
+    interrupt = 3
+
+    @composite
+    def interrupt_eventually(draw):
+        runs[0] += 1
+        if runs[0] == interrupt:
+            # import pdb; pdb.set_trace()
+            raise KeyboardInterrupt
+        return draw(integers())
+
+    @given(interrupt_eventually())
+    def test_do_nothing(x):
+        pass
+
+    with pytest.raises(KeyboardInterrupt):
+        test_do_nothing()
+
+    assert runs[0] == interrupt
+
+
+def test_systemexit_no_rerun():
```

Comment:
Would it make sense to merge this with `test_keyboardinterrupt_no_rerun` using `pytest.mark.parametrize`?

Reply:
Makes sense to me - we could also test raising `SystemExit` from a strategy.

Reply:
I'm not totally sure what behaviour we want from SystemExit, it's a fairly weird exception to ever raise; this was more relevant when I was worrying about "all' BaseExceptions.

Reply:
Actually while I'm clear what KeyboardInterrupt should do I don't really have any idea what SystemExit or anything else should do. Right now SystemExit raised in the system under test propagates out through the `given` decorator and is caught further up; you don't get a falsifying example or a Flaky if it arises part of the time (as you would with ValueError. If raised in a strategy, SystemExit behaves like ValueError. I'll check in a version that documents what's going on now; maybe don't merge it until it's clear we're doing the right thing.